### PR TITLE
bitECS: Fix move permissions

### PR DESCRIPTION
**Pull request by keianhzo**
Fixes https://github.com/mozilla/hubs/issues/6250

This PR fixes the moving permission for media.

### CODE REVIEW

File: src/utils/permissions-utils.js [hunk 1]
```diff
@@ -18,22 +19,31 @@ export function showHoverEffect(el) {
...
-  const isEmoji = !!entity.components.emoji;
-  return (
-    hasComponent(APP.world, HoldableButton, entity.eid) ||
-    ((isEmoji || isEmojiSpawner
-      ? window.APP.hubChannel.can("spawn_emoji")
-      : window.APP.hubChannel.can("spawn_and_move_media")) &&
-      (!isPinned || window.APP.hubChannel.can("pin_objects")) &&
-      (!isPen || window.APP.hubChannel.can("spawn_drawing")))
-  );
+export function canMove(elOrEid) {
```
[the remark this answers is not in the document]
Comment (keianhzo) [reply src/utils/permissions-utils.js]: The one thing I like about the elOrEid approach  is that avoids branching in a lot of places when the function is used across the codebase. In this case `canMove` is only used in a couple of places so I think it's fine to duplicate the function.

**Comment by keianhzo** [comment]
@takahirox I've separated the functions and also re-enable a commented canMove in `app-aware-touchscreen.js`.

File: src/utils/bit-permissions-utils.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { hasComponent } from "bitecs";
+import { HoldableButton } from "../bit-components";
+import { isPinned } from "../bit-systems/networking";
+
+export function canMove(eid) {
+  // Add support to pen/emojis when those are migrated to bitECS
+  return (
+    hasComponent(APP.world, HoldableButton, eid) ||
+    (APP.hubChannel.can("spawn_and_move_media") && (!isPinned(eid) || APP.hubChannel.can("pin_objects")))
+  );
```
Comment (takahirox) [inline src/utils/bit-permissions-utils.js:10]: Review note: Honestly I'm not really familiar well enough permission check but it looks based on `utils/permission-utils.js` so I want to approve for now.

**Comment by takahirox** [comment]
> re-enable a commented canMove in app-aware-touchscreen.js.

Is this related to #6250?

**Review by takahirox — APPROVED** [review]
Approves once https://github.com/mozilla/hubs/pull/6253#issuecomment-1736747401 will be clear

**Comment by keianhzo** [comment]
@takahirox yes, it's related to that, otherwise you could move pinned objects.

**Comment by takahirox** [comment]
OK, good to go

**Pull request merged**